service/history/historyEngine.go

```diff
@@ -3273,6 +3273,9 @@ func validateStartChildExecutionAttributes(parentInfo *persistence.WorkflowExecu
 }
 
 func validateStartWorkflowExecutionRequest(request *workflow.StartWorkflowExecutionRequest, maxIDLengthLimit int) error {
+	if request.RequestId == nil || len(common.StringDefault(request.RequestId)) == 0 {
```

Comment:
I think checking len(request.requestID) is sufficient.

Reply:
updated. Thanks!